internal/logic/logic_test.go

```diff
@@ -285,10 +287,254 @@ func TestTransactionManagementServiceLogic_NewTransaction(t *testing.T) {
+				mockDs := mock.NewMockDataSourceI(mockCtrl)
+				var trans []model.Transaction
+				trans = append(trans, model.Transaction{UserId: "123", AccountNumber: 1})
+				mockDs.EXPECT().Get(map[string]interface{}{"transaction_id": "123"}, 0, 0).Times(1).Return(trans, 1, nil)
+				router := mux.NewRouter()
+				router.HandleFunc("/microbank/v1/user", func(w http.ResponseWriter, r *http.Request) {
+					json.NewEncoder(w).Encode(Reader(""))
+				})
+				mockPdf := pdfMock.NewMockHtmlToPdfSvcI(mockCtrl)
+				srv := httptest.NewServer(router)
+				return mockDs, config.ExternalSvc{UserSvc: srv.URL, PdfSvc: config.PdfSvc{UuId: "11-22-33-44", PdfService: mockPdf}}
+			},
+			want: func(resp *respModel.Response) {
+				temp := respModel.Response{
+					Status:  http.StatusInternalServerError,
+					Message: codes.GetErr(codes.ErrUnmarshall),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(resp, &temp) {
+					t.Errorf("Want: %v, Got: %v", temp, &resp)
+				}
+			},
+		},
+		{
+			name:          "Failure :: DownloadPdf :: error assert response data",
+			transactionId: "123",
+			setup: func() (datasource.DataSourceI, config.ExternalSvc) {
+				mockDs := mock.NewMockDataSourceI(mockCtrl)
+				var trans []model.Transaction
+				trans = append(trans, model.Transaction{UserId: "123", AccountNumber: 1})
+				mockDs.EXPECT().Get(map[string]interface{}{"transaction_id": "123"}, 0, 0).Times(1).Return(trans, 1, nil)
+				router := mux.NewRouter()
+				router.HandleFunc("/microbank/v1/user", func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusOK, "Success", Reader(""))
+				})
+				mockPdf := pdfMock.NewMockHtmlToPdfSvcI(mockCtrl)
+				srv := httptest.NewServer(router)
+				return mockDs, config.ExternalSvc{UserSvc: srv.URL, PdfSvc: config.PdfSvc{UuId: "11-22-33-44", PdfService: mockPdf}}
+			},
+			want: func(resp *respModel.Response) {
+				temp := respModel.Response{
+					Status:  http.StatusInternalServerError,
+					Message: codes.GetErr(codes.ErrAssertResp),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(resp, &temp) {
+					t.Errorf("Want: %v, Got: %v", temp, &resp)
+				}
+			},
+		},
+		{
+			name:          "Failure :: DownloadPdf :: error generate pdf",
+			transactionId: "123",
+			setup: func() (datasource.DataSourceI, config.ExternalSvc) {
+				mockDs := mock.NewMockDataSourceI(mockCtrl)
+				var trans []model.Transaction
+				trans = append(trans, model.Transaction{UserId: "123", AccountNumber: 1})
+				mockDs.EXPECT().Get(map[string]interface{}{"transaction_id": "123"}, 0, 0).Times(1).Return(trans, 1, nil)
+				router := mux.NewRouter()
+				router.HandleFunc("/microbank/v1/user", func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusOK, "Success", map[string]interface{}{"Name": "abc"})
+				})
+				mockPdf := pdfMock.NewMockHtmlToPdfSvcI(mockCtrl)
+				mockPdf.EXPECT().GeneratePdf(gomock.Any(), gomock.Any()).Return([]byte("PDF"), errors.New("pdf generate error"))
```

Comment:
Is the data to this mock unpredictable? If it's not unpredictable, replace gomock.Any() with the exact expected data.